src/Network/Network/VirtualNetwork/Subnet/GetAzureVirtualNetworkSubnetConfigCommand.cs

```diff
@@ -17,44 +17,69 @@
 using System.Linq;
 using System.Management.Automation;
 using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+using Microsoft.Azure.Commands.Network.Common;
 
 namespace Microsoft.Azure.Commands.Network
 {
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkSubnetConfig"), OutputType(typeof(PSSubnet))]
-    public class GetAzureVirtualNetworkSubnetConfigCommand : NetworkBaseCmdlet
-    {
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkSubnetConfig",DefaultParameterSetName = "GetByVirtualNetwork"), OutputType(typeof(PSSubnet))]
+    public class GetAzureVirtualNetworkSubnetConfigCommand : VirtualNetworkBaseCmdlet
+  {
         [Parameter(
             Mandatory = false,
+            ParameterSetName = "GetByVirtualNetwork",
             HelpMessage = "The name of the subnet")]
         public string Name { get; set; }
 
         [Parameter(
             Mandatory = true,
             ValueFromPipeline = true,
+            ParameterSetName = "GetByVirtualNetwork",
             HelpMessage = "The virtualNetwork")]
         public PSVirtualNetwork VirtualNetwork { get; set; }
 
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "GetByResourceId",
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Id to the subnet"
+        )]
+        public string ResourceId { get; set; }
+
         public override void Execute()
         {
             base.Execute();
-            if (!string.IsNullOrEmpty(this.Name))
-            {
+
+            if (ParameterSetName == "GetByResourceId") {
+              string resourceGroupName = null;
+              string virtualNetworkName = null;
+              string subnetName = null;
+
+              if (ResourceIdHelper.TryParseSubnetMetadataFromResourceId(this.ResourceId, out resourceGroupName, out virtualNetworkName, out subnetName)) {
+                this.Name = subnetName;
+                this.VirtualNetwork = this.GetVirtualNetwork(resourceGroupName, virtualNetworkName);
+              }
+            }
+
+            if (this.VirtualNetwork != null) {
```

Comment:
Please indent with four space to keep consistent code style

Reply:
Done